tests/regression/test_interp_dual.py

```diff
@@ -0,0 +1,168 @@
+import pytest
+import numpy as np
+from firedrake import *
+import ufl
+
+
+@pytest.fixture(scope='module')
+def mesh():
+    return UnitSquareMesh(5, 5)
+
+
+@pytest.fixture
+def M(fs):
+    uhat = TrialFunction(fs)
+    v = TestFunction(fs)
+    return inner(uhat, v) * dx
+
+
+def test_assemble_interp(mesh):
+    V1 = FunctionSpace(mesh, "CG", 1)
+    V2 = FunctionSpace(mesh, "CG", 2)
+
+    x, y = SpatialCoordinate(mesh)
+    expr = cos(2*pi*x)*sin(2*pi*y)
+    f1 = Function(V1).interpolate(expr)
+
+    # Check type
+    If1 = Interp(f1, V2)
+    assert isinstance(If1, ufl.Interp)
+
+    # -- I(f1, V2) -- #
+    a = assemble(If1)
+    b = interpolate(f1, V2)
+    assert np.allclose(a.dat.data, b.dat.data)
+
+    assembled_If1 = assemble(If1)
+    assert np.allclose(assembled_If1.dat.data, b.dat.data)
```

Comment:
It seems that lines 36-37 are the same as lines 32 and 34.

Reply:
Thank you very much! I am not sure what happened here.